Dockerfile.api

```diff
@@ -1,8 +1,15 @@
-FROM python:3.11
-
-WORKDIR /sqlmesh
-
+FROM python:3.11-slim-bullseye AS venv-image
+RUN apt-get update && apt-get -y install libpq-dev gcc
 COPY setup.py setup.py
 COPY sqlmesh/_version.py sqlmesh/_version.py
+ENV VIRTUAL_ENV=/.venv
+ENV PATH="$VIRTUAL_ENV/bin:$PATH"
+RUN python3 -m venv $VIRTUAL_ENV
+RUN pip3 install -e .[dev,web]
 
-RUN pip install -e .[dev,web]
+FROM python:3.11-slim-bullseye as app-image
```

Comment:
why are you referencing python 3.11 both times? can you explain a bit about what this pr is doing?

Reply:
The Dockerfile was updated the following way:
1) The "slimmer" base image `python:3.11-slim-bullseye` was used.
2) Multi-stage build approach was used: https://docs.docker.com/build/building/multi-stage

The #2 is why the image is being referenced twice. Hope that helps! Let me know if you need more clarity on this ;)